internal/sign/manager.go

```diff
@@ -0,0 +1,26 @@
+package sign
+
+import (
+	"context"
+
+	kmmv1beta1 "github.com/kubernetes-sigs/kernel-module-management/api/v1beta1"
+)
+
+type Status string
+
+const (
+	StatusCompleted  = "completed"
+	StatusCreated    = "created"
+	StatusInProgress = "in progress"
+)
+
+type Result struct {
+	Requeue bool
+	Status  Status
+}
+
+//go:generate mockgen -source=manager.go -package=sign -destination=mock_manager.go
+
+type SignManager interface {
+	Sync(ctx context.Context, mod kmmv1beta1.Module, m kmmv1beta1.KernelMapping, targetKernel string, previousImage string, targetImage string, pushImage bool) (Result, error)
```

Comment:
What is `previousImage`?

Reply:
i think it should be renamed to "imageToSign"